### DISCUSSION TITLE
Vaultwarden export stops after few entries

### DISCUSSION THREAD

**Discussion by nakal** [dani-garcia/vaultwarden#1919] [Q&A]
### Subject of the issue

The function Tools / Vault export is broken for me. It stops after a few entries (here: 13) and is incomplete. I have got ~550 entries.

### Deployment environment

* FreeBSD 12.2
* nginx
* vaultwarden version: 1.22.2

* Install method: OS package (compiled with Poudiere)
* Clients used: Firefox on FreeBSD
* Reverse proxy and version: nginx 1.20.1
* MySQL/MariaDB or PostgreSQL version: MariaDB 10.4.20

* More information: Vaultwarden was upgraded from Bitwarden manually by following the documentation.

### Steps to reproduce

1. Go to the web interface of Vaultwarden using your web browser. (Log in, if needed.)
2. Click on "Tools".
3. Click on Export, use CSV or JSON, enter master password.
4. See downloaded file. It has fewer entries.

### Expected behaviour

It should export the full vault contents.

### Actual behaviour

It stops at some point, just after few passwords.

### Troubleshooting data

Nothing of use in logs (nginx-error.log or vaultwarden.log).

It could be some encoding problems (I use probably use typical German characters somewhere). The last exported entry actually uses a German word with umlauts "Sicherungsschlüssel".

**Comment by BlackDex** [comment]
That is not something done by Vaultwarden. The export is done fully within the clients since they have all the logic regarding encryption and decryption.

That is also the reason why there are no useful logs within Vaultwarden or your Reverse Proxy.

**Reply by nakal** [reply]
So what do I do to export all my password entries to CSV, if the function runs but then stops in the middle of the export? You mean it is a bug in Firefox/Chrome (I tried both clients now) and not Vaultwarden?

Or what do you mean by "client"? Is Vaultwarden divided in client and server components? Did I report the bug to the wrong repository?

Edit: I updated the steps to reproduce the bug. Maybe it helps.

**Reply by jjlin** [reply]
Typically, "incomplete" exports are due to shared (organization-owned) items. You may be seeing 500 entries, but `Export Vault` only exports the entries you own directly. If you want the shared items, you need to go into each org view and export the org vault separately.

**Reply by nakal** [reply]
I tried it, but the export is not in the context of any view. It is a separate submenu under "Tools". I can also see that the export contains multiple folders. I own 515 password entries *directly*.

Update:
I clicked on an organisation and there is actually a second Tools menu that appears under the main Tools menu. I'll test it and report if the export is successful. Thanks for the hint.

**Reply by nakal** [reply]
I tested it. It seems you need to select the organisations one by one and then additionally the own main export.

The UI is not very intuitive, in my opinion. In the main view you can see all passwords, but can export only the ones that do not belong to an organisation. Then, there are also two menus for exporting logins. One top menu, the other context sensitive that you can see only if you select an organisation. The usability is also not very good at this point.

Thank you very much for the help.

**Reply by BlackDex** [reply]
There is an other option you could try to use: https://github.com/vwxyzjn/portwarden
It isn't updated for a while, but it could work for you. Though importing will not create orgs or collections if i'm correct.
And, the benefit is, it will also backup your attachments.
